common-util/src/test/java/org/sunbird/common/models/util/BaseForHttpTest.java

```diff
@@ -0,0 +1,85 @@
+package org.sunbird.common.models.util;
+
+import static org.powermock.api.mockito.PowerMockito.mock;
+import static org.powermock.api.mockito.PowerMockito.when;
+import static org.powermock.api.mockito.PowerMockito.whenNew;
+import static org.powermock.api.mockito.PowerMockito.doThrow;
+
+import java.io.BufferedReader;
+import java.io.FileNotFoundException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.io.OutputStreamWriter;
+import java.net.HttpURLConnection;
+import java.net.URL;
+
+import javax.management.Notification;
+
+import org.apache.http.impl.client.HttpClients;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.runner.RunWith;
+import org.mockito.Mockito;
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.core.classloader.annotations.PowerMockIgnore;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+import org.sunbird.services.sso.impl.KeyCloakServiceImpl;
+
+@RunWith(PowerMockRunner.class)
+@PowerMockIgnore({"javax.management.*", "javax.net.ssl.*", "javax.security.*"})
+@PrepareForTest({
+  OutputStreamWriter.class,
+  URL.class,
+  BufferedReader.class,
+  HttpUtil.class,
+  HttpClients.class,
+  Notification.class,
+  KeyCloakConnectionProvider.class,
+  KeyCloakServiceImpl.class
+})
+public abstract class BaseForHttpTest {
+
+  @Before
+  public void addMockRules() {
+
+    httpRules("content/v3/list", "not-empty-output");
+    httpRules("/search/health", "not-empty-output");
+    httpRules("/content/wrong/v3/list", null, true, null);
+    httpRules("v1/issuer/issuers", "{\"message\":\"success\"}");
+    httpRules("https://dev.ekstep.in/api/data/v3", "{\"message\":\"success\"}");
+  }
+
+  protected void httpRules(String urlContains, String outputexpected) {
+    httpRules(urlContains, outputexpected, false, null);
+  }
+
+  protected void httpRules(
+      String urlContains, String outputexpected, boolean throwError, String paramContains) {
```

Comment:
Is paramContains used? In all cases, it seems to be only passed null value.

Reply:
Used from sub-class only - current methods usage defaults to null